spec/controllers/api/orders_controller_spec.rb

```diff
@@ -156,6 +158,109 @@ module Api
+          get :show, id: order.number
+          expect_order
+        end
+
+        it "returns the order, as the order distributor owner" do
+          allow(controller).to receive(:spree_current_user) { order.distributor.owner }
+          get :show, id: order.number
+          expect_order
+        end
+
+        it "returns unauthorized, as the order product's supplier owner" do
+          allow(controller).to receive(:spree_current_user) { order.line_items.first.variant.product.supplier.owner }
+          get :show, id: order.number
+          assert_unauthorized!
+        end
+
+        it "returns the order, as the Order Cycle coorinator owner" do
+          allow(controller).to receive(:spree_current_user) { order.order_cycle.coordinator.owner }
+          get :show, id: order.number
+          expect_order
+        end
+      end
+
+      context "as distributor owner" do
+        let!(:order) { create(:completed_order_with_fees, order_cycle: order_cycle, distributor: distributor ) }
+
+        before { allow(controller).to receive(:spree_current_user) { order.distributor.owner } }
+
+        it "can view an order not in a standard state" do
+          order.update_attributes(completed_at: nil, state: 'shipped')
+          get :show, id: order.number
+          expect_order
+        end
+
+        it "returns an order with all required fields" do
+          get :show, id: order.number
+          expect_order
+          expect_detailed_attributes_to_be_present(json_response)
+
+          expect(json_response[:bill_address][:address1]).to eq(order.bill_address.address1)
+          expect(json_response[:bill_address][:lastname]).to eq(order.bill_address.lastname)
+          expect(json_response[:ship_address][:address1]).to eq(order.ship_address.address1)
+          expect(json_response[:ship_address][:lastname]).to eq(order.ship_address.lastname)
+          expect(json_response[:shipping_method][:name]).to eq(order.shipping_method.name)
+          json_response[:adjustments].each do |adjustment|
+            if adjustment[:label] == "Transaction fee"
+              expect(adjustment[:amount]).to eq(order.adjustments.payment_fee.first.amount.to_s)
+            elsif json_response[:adjustments].first[:label] == "Shipping"
+              expect(adjustment[:amount]).to eq(order.adjustments.shipping.first.amount.to_s)
+            end
+          end
+          expect(json_response[:payments].first[:amount]).to eq(order.payments.first.amount.to_s)
+          expect(json_response[:line_items].size).to eq(order.line_items.size)
+          expect(json_response[:line_items].first[:variant][:product_name]). to eq(order.line_items.first.variant.product.name)
+        end
+      end
+
+      def expect_order
+        expect(response.status).to eq 200
+        expect_correct_order(json_response, order)
+      end
+
+      def expect_correct_order(json_response, order)
+        expect(json_response[:number]).to eq(order.number)
```

Comment:
isn't the number enough? it's a unique way to identify it.

Reply:
yes, ok, I removed the email :+1: